### PR TITLE
refactoring aptrace

### PR DESCRIPTION
**Pull request by YuiKasagi**
I modified aptrace.py so that it can be applied to REACH data and YJ band.
(I think this would probably work for a different data set if "cutrow" is adjusted.)
The current version only accepts nap=42 (for H band) or 102 (for YJ band).

### CODE REVIEW

File: examples/python/REACH_stream.py [hunk 1]
```diff
@@ -5,8 +5,21 @@
...
 
+# aperture extraction
+datadir = pathlib.Path('/home/kawahara/pyird/data/flat/')
+anadir = pathlib.Path('/home/kawahara/pyird/data/flat/')
+flat_smf66=irdstream.Stream2D("flat_smf66",datadir,anadir)
+flat_smf66.fitsid=list(range(47224,47246,2)) #no light in the speckle fiber
+flat_smf66.fitsid_increment()
+flatmedian=flat_smf66.immedian()
+cutrow = 1000  ## 501 ~ 1549
+nap = 102 ## 42 for H band, 102 for YJ band
```
Comment (HajimeKawahara) [inline examples/python/REACH_stream.py:19]: 42?

File: src/pyird/image/aptrace.py [hunk 2]
```diff
@@ -3,26 +3,41 @@
...
+from pyird.plot.order import plot_crosssection, plot_tracelines
 
-def cross_section(dat,nrow):
-    """extract cross section in row direction.
+def cross_section(dat,nrow,nap):
+    """extract cross section in row direction and search peaks.
 
     Args:
         dat: flat data
         nrow: row number to be extracted
```
Comment (HajimeKawahara) [inline src/pyird/image/aptrace.py:16]: Add an explanation of `nap`.

File: src/pyird/image/aptrace.py [hunk 3]
```diff
@@ -37,17 +52,28 @@ def set_aperture(dat,cutrow,nap):
...
+            prange = (peakind_cut[-1]>1500) or (peakind_cut[1]>40)
+        elif nap>100: # yj band
+            prange = (peakind_cut[0]<250) or (peakind_cut[-2]<2000)
+        cutrow_lim = (cutrow>cutrow_min) and (cutrow<cutrow_max)
         cutrow += 1
-
+    if not cutrow_lim:
+        print('Error: %d apertures could not be found.'%(nap))
+        print('Please change "cutrow" or "nap" or confirm the orientation of the image.')
+        sys.exit()
```
Comment (HajimeKawahara) [inline src/pyird/image/aptrace.py:71]: Considering the case one runs the code in Jupyter notebook, `return` is better?
If you agree, remove `import sys` too.

File: src/pyird/image/aptrace.py [hunk 4]
```diff
@@ -171,14 +205,20 @@ def aptrace(dat,cutrow,nap):
     Returns:
         parameters of a polynomial to trace apertures
     """
-    peakind_cut = set_aperture(dat,cutrow,nap)
+    if nap!=42 and nap!=102:
+        print('Please set nap = 42 (for H band) or nap = 102 (for YJ band).')
+        sys.exit()
```
Comment (HajimeKawahara) [inline src/pyird/image/aptrace.py:210]: same as the above one.

File: src/pyird/plot/order.py [hunk 5]
```diff
@@ -39,3 +39,12 @@ def plot_crosssection(pixels,flux,peakind):
     ax.plot(pixels[peakind],flux[peakind],'x')
 
     plt.show()
+
+def plot_tracelines(x,y):
+    fig=plt.figure()
```
Comment (HajimeKawahara) [inline src/pyird/plot/order.py:44]: Add `Args` and `Returns`.

**Review by HajimeKawahara — APPROVED** [review]
Thanks! 
`python REACH_stream.py`
worked when modifying [this](https://github.com/prvjapan/pyird/pull/33/files#r846886665).

File: src/pyird/image/aptrace.py (hunk 3 above)
Comment (YuiKasagi) [reply src/pyird/image/aptrace.py:71]: I agree. I will change.

**Comment by HajimeKawahara** [comment]
Thanks for update!

**Pull request merged**